caddyhttp/basicauth/basicauth_test.go

```diff
@@ -14,19 +14,31 @@ import (
 )
 
 func TestBasicAuth(t *testing.T) {
+	var i int
+	// This handler is registered for tests in which the only authorized user is
+	// "okuser"
+	upstreamHandler := func(w http.ResponseWriter, r *http.Request) (int, error) {
+		remoteUser, _ := r.Context().Value("remote_user").(string)
```

Comment:
The more I think about it, the net result is the same. It's not technically an error value, it's a boolean telling you whether the assertion was successful. I like the brevity of your approach. I just forgot that type assertions don't panic if you assign to two variables, even if one is discarded. This is fine.